### DISCUSSION TITLE
Allow the second parameter of the assign function to accept 'undefined' and 'null'

### DISCUSSION THREAD

**Discussion by moo-w** [radashi-org/radashi#117] [Ideas]
```ts
let obj

Object.assign({ a: 'b' }, obj) // { a: 'b' }

obj = null

Object.assign({ a: 'b' }, obj) // { a: 'b' }
```

I think maybe we should allow the second parameter of the assign function to accept 'undefined' and 'null' just like `Object.assign`.

**Comment by moo-w** [comment]
https://github.com/radashi-org/radashi/blob/a1349e9e35ae415d73776a8e2a085d873df23e87/src/object/assign.ts#L17-L20

Change to 👇

```ts
export function assign<X extends Record<string | symbol | number, any>>(
  initial: X,
  override?: X | null,
): X {}
```
or
```ts
export function assign<X extends Record<string | symbol | number, any>>(
  initial?: X | null,
  override?: X | null,
): X {}
```

**Reply by aleclarson** [reply]
`Object.assign` doesn't allow the first argument to be null or undefined.

```
Uncaught TypeError: Cannot convert undefined or null to object
    at Function.assign (<anonymous>)
```

Also, `Object.assign` supports more than 2 arguments. Should `assign` do the same?

**Reply by aleclarson** [reply]
Hi @MoozonWei, it's been 3 weeks since I've heard from you. Did you understand my reply?
